packages/core/src/string.ts

```diff
@@ -60,3 +60,61 @@ String.prototype.startsWith = function (
     if (!(position > 0)) position = 0
     return this.indexOf(searchString, position, position + 1) >= 0
 }
+
+String.prototype.split = function (
+    this: string,
+    separator?: string,
+    limit?: number
+): string[] {
+    const S = this
+    // https://www.ecma-international.org/ecma-262/6.0/#sec-string.prototype.split
+    const A: string[] = []
+    let lim = 0
+    if (limit === undefined) lim = (1 << 29) - 1
+    // spec says 1 << 53, leaving it at 29 for constant folding
+    else if (limit < 0) lim = 0
+    else lim = limit | 0
+    const s = S.length
+    let p = 0
+    const R = separator
+    if (lim === 0) return A
+    if (separator === undefined) {
+        A[0] = S
+        return A
+    }
+    if (s === 0) {
+        let z = splitMatch(S, 0, R)
+        if (z > -1) return A
+        A[0] = S
+        return A
+    }
+    let T: string
+    let q = p
+    while (q !== s) {
+        let e = splitMatch(S, q, R)
+        if (e < 0) q++
+        else {
+            if (e === p) q++
+            else {
+                T = S.slice(p, q)
+                A.push(T)
+                if (A.length === lim) return A
+                p = e
+                q = p
+            }
+        }
+    }
+    T = S.slice(p, q)
+    A.push(T)
+    return A
+}
+
+function splitMatch(S: string, q: number, R: string): number {
```

Comment:
isn't this `S.indexOf(R, q)` ?

Reply:
more like indexOf(R, q) === q ? q + r : -1

Reply:
oh, I see; we have an additional argument to indexOf() so you can do `indexOf(R, q, q + 1) === q ? q + r : -1`, see https://github.com/microsoft/devicescript/blob/8b6f5a971a08ed61ee2055be45899a3d1b476849/packages/core/src/corelib.d.ts#L129